src/PrestaShopBundle/Controller/Admin/Sell/Catalog/AttributeController.php

```diff
@@ -0,0 +1,217 @@
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Controller\Admin\Sell\Catalog;
+
+use PrestaShop\PrestaShop\Core\Domain\Attribute\Command\BulkDeleteAttributeCommand;
+use PrestaShop\PrestaShop\Core\Domain\Attribute\Command\DeleteAttributeCommand;
+use PrestaShop\PrestaShop\Core\Domain\Attribute\Exception\AttributeException;
+use PrestaShop\PrestaShop\Core\Domain\Attribute\Exception\AttributeNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Attribute\Exception\DeleteAttributeException;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\Exception\AttributeGroupException;
+use PrestaShop\PrestaShop\Core\Grid\Definition\Factory\AttributeGridDefinitionFactory;
+use PrestaShop\PrestaShop\Core\Search\Filters\AttributeFilters;
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+use PrestaShopBundle\Security\Annotation\AdminSecurity;
+use PrestaShopBundle\Service\Grid\ResponseBuilder;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Responsible for Sell > Catalog > Attributes & Features > Attributes > Attribute
+ */
+class AttributeController extends FrameworkBundleAdminController
+{
+    /**
+     * Displays Attribute groups > attributes page
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_attributes_index",
+     *     redirectQueryParamsToKeep={"attributeGroupId"}
+     * )
+     *
+     * @param Request $request
+     * @param $attributeGroupId
+     * @param AttributeFilters $attributeFilters
+     *
+     * @return Response
+     */
+    public function indexAction(Request $request, $attributeGroupId, AttributeFilters $attributeFilters)
+    {
+        try {
+            $attributeGridFactory = $this->get('prestashop.core.grid.factory.attribute');
+            $attributeGrid = $attributeGridFactory->getGrid($attributeFilters);
+
+            return $this->render('@PrestaShop/Admin/Sell/Catalog/Attribute/index.html.twig', [
+                'attributeGrid' => $this->presentGrid($attributeGrid),
+                'attributeGroupId' => $attributeGroupId,
+                'enableSidebar' => true,
+                'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+            ]);
+        } catch (AttributeGroupException $e) {
```

Comment:
Shouldn't you catch not found exception only?

Reply:
It might throw NotFound and in extra rare cases generic AttributeGroupException (the one thrown instead of PrestashopException if something goes wrong with db)

Reply:
Why would you hide error when something unexpected happens?

Reply:
Agreed with @sarjon 
We must catch the Exceptions that we expect can happen in a normal usage in order to display a useful error message. Basically we expect these Exceptions to be thrown because of a bad usage of the application, not because something went wrong inside the software.

Normal usage:
- attempt to load a attribute using a bad ID => NotFoundException
- attempt to persist ObjectModel but get rejected by some constraint
- attempt to send an email, fails because SMTP settings were badly inserted

But if database times out, we can show a standard error page as the software went wrong.

The idea is really:
- if an Exception is thrown because of user doing something bad/stupid/crazy/unlucky* catch it to display nice error message
- if an Exception is thrown because of something wrong, dont catch it, let the standard error message happen

*Classic example: open 2 tabs of the same product in BO, delete in tab 1 then refresh tab 2.

Reply:
@matks, sorry I am still missing some parts :smile: .  Can you clarify what do you mean with `standard error message` ? 1. Is it when we throw `An unexpected error occurred. [%type% code %code%]` or 2. default behavior of error, like `error 500` ?
Because if it's the first option, then it still requires the exception to be caught in controller, else the second option happens (But then, what's the point catching PrestashopException and converting it to domain exception in all handlers if it's not supposed to be caught anyway?). Thanks.